language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/RightArrowActionNodeContext.java

```diff
@@ -55,34 +50,14 @@ public RightArrowActionNodeContext(Class<T> attachmentPoint) {
     protected List<LSCompletionItem> getFilteredItems(BallerinaCompletionContext context, ExpressionNode node) {
         List<LSCompletionItem> completionItems = new ArrayList<>();
         List<Symbol> visibleSymbols = context.visibleSymbols(context.getCursorPosition());
-        Predicate<Symbol> predicate;
-        switch (node.kind()) {
-            case SIMPLE_NAME_REFERENCE:
-                predicate = symbol -> symbol.kind() != SymbolKind.FUNCTION
-                        && Objects.equals(symbol.getName().orElse(null),
-                                          ((SimpleNameReferenceNode) node).name().text());
-                break;
-            case FUNCTION_CALL:
-                predicate = symbol -> symbol.kind() == SymbolKind.FUNCTION
-                        && Objects.equals(symbol.getName().orElse(null),
-                                          ((SimpleNameReferenceNode) ((FunctionCallExpressionNode) node)
-                                                  .functionName()).name().text());
-                break;
-            default:
-                return completionItems;
-        }
-
-        Optional<Symbol> filteredEntry = visibleSymbols.stream().filter(predicate).findAny();
-
-        if (filteredEntry.isEmpty()) {
-            return completionItems;
-        }
-        if (SymbolUtil.isClient(filteredEntry.get())) {
+        Optional<TypeSymbol> expressionType = context.currentSemanticModel().orElseThrow().type(node);
```

Comment:
Instead of throwing, we can do `flatMap()` as below:
```suggestion
        Optional<TypeSymbol> expressionType = context.currentSemanticModel().flatMap(semanticModel -> semanticMode.type(node));
```
WDYT? IMO, this approach is much cleaner.